nexus/db-queries/src/db/datastore/vmm.rs

```diff
@@ -39,8 +39,13 @@ use uuid::Uuid;
 
 /// The result of an [`DataStore::vmm_and_migration_update_runtime`] call,
 /// indicating which records were updated.
-#[derive(Copy, Clone, Debug)]
+#[derive(Clone, Debug)]
 pub struct VmmStateUpdateResult {
+    /// The VMM record that the update query found and possibly updated.
+    ///
+    /// NOTE: This is the record prior to the update!
+    pub found_vmm: Vmm,
```

Comment:
Take it or leave it: If we're concerned about people potentially misusing this, we *could* return just the ID, since it's immutable, and that's the part we actually need?

Reply:
Alternatively...this query is no longer a CTE, but a bunch of smaller queries in a transaction. So, we _could_ just update the VMM record with a `RETURNING` clause and, if it fails, just do another query to fetch the current state? I dunno, that's more queries...

Reply:
I think I'm comfortable having the whole record in here--we're pretty clear throughout the code base that the conditional-update routines generally don't return updated values--but I figured I'd leave a note for myself anyway. We can amend the transaction to return the updated record later on if we run into a case where we need it.